config/rollup.config.js

```diff
@@ -5,117 +5,125 @@ import replace from 'rollup-plugin-replace'
 import commonjs from 'rollup-plugin-commonjs'
 import { terser } from 'rollup-plugin-terser'
 import { sizeSnapshot } from 'rollup-plugin-size-snapshot'
+import esbuild from 'rollup-plugin-esbuild' // Used for TS transpiling
```

Comment:
@gdorsi I can't really say **why** it was added in the first place but there are many sources out there showing that `esbuild` is, indeed, quite faster than `babel` for this job.

A few sources are:
- https://github.com/evanw/esbuild
- https://github.com/privatenumber/esbuild-loader/discussions/138
- https://datastation.multiprocess.io/blog/2021-11-13-benchmarking-esbuild-swc-typescript-babel.html

Far from being a benchmark in itself, I've run some quick tests locally and building `graphql-hooks` with `babel` takes ~6 seconds on my machine. With `esbuild` it takes ~4 seconds.

-----

That aside, any particular issue why you think `babel` should be used instead?

Reply:
We are already using babel for the es5 compilation, so to support the typescript compilation would be enough to add the typescript plugin.

More than solving a real problem, I was curious if adding the typescript plugin on babel costs us more than adding esbuild.

Thanks for checking!

Reply:
The main reason I added esbuild is that babel isn't used everywhere in this. one of the build types doesn't use babel and I assumed this was intentional to provide non transpiled files.

So instead of adding babel on things that didn't have it, I just make everything go through esbuild to strip the typescript.

I also remember having another specific issue when I tried with babel, so I ended up switching to esbuild which solves the problem easily without issues and compiles fast

There's no documentation on why the build was was like that in the first place so it's hard to tell what's the best thing to do, but the current setup works